music-ecom/src/Context/AuthContext.js

```diff
@@ -0,0 +1,40 @@
+import { createContext, useContext, useEffect, useReducer } from "react";
+import AuthReducerFunc from "../Reducers/AuthReducer";
+
+const AuthInitialState = {
+  token: null,
+  IsLoggedIn: false,
+};
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [AuthState, AuthDispatch] = useReducer(
+    AuthReducerFunc,
+    AuthInitialState
+  );
+  useEffect(() => {
+    function oneTime(AuthDispatch) {
+      console.log("I am here in USEEFFECT");
+      if (localStorage.getItem("loginToken" !== null)) {
+        console.log("Login is continuing");
+        return AuthDispatch({ type: "LOGIN_PERSIST" });
+      } else {
+        console.log("Masla hgya ji USEEFFECT");
```

Comment:
Interesting console log messages .. But better to toastify it with sensible alerts :)

Reply:
ok😅